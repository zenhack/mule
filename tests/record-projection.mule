#XFAIL: type formatting
{x = {}}.x
