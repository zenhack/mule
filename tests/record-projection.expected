
Desugared:

	(. (letrec (types ()) (values ((x {}))) (((.= x) {}) x)) x)

inferred type:

	(record (types ()) (values ()))

Runtime term:

	(App
	 ((GetField x)
	  (LetRec
	   (((0 (Record ())))
	    (App
	     ((App
	       ((Lam
	         (0 () (Lam (1 () (Update (old (Var 1)) (label x) (field (Var 0)))))))
	        (Record ())))
	      (Var 0)))))))

Evaluated:

	(Record ())
